File: app/src/main/java/wangdaye/com/geometricweather/location/services/ip/BaiduIPLocationApi.java
package wangdaye.com.geometricweather.location.services.ip;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface BaiduIPLocationApi {

    @GET("location/ip")
    Observable<BaiduIPLocationResult> getLocation(@Query("ak") String ak,
                                                  @Query("coor") String coor);
}

File: app/src/main/java/wangdaye/com/geometricweather/location/services/ip/BaiduIPLocationApi.kt
package wangdaye.com.geometricweather.location.services.ip

import retrofit2.http.GET
import retrofit2.http.Query

interface BaiduIPLocationApi {

    @GET("location/ip")
    suspend fun getLocation(
            @Query("ak") ak: String,
            @Query("coor") coor: String
    ): BaiduIPLocationResult
}